### DISCUSSION TITLE
Menu Bar Issue

### DISCUSSION THREAD

**Discussion by SteveMattan** [ttscoff/bunch-docs#177] [General]
I do not know that this is an issue with Bunch or Bartender or something else.

Here is an issue I have encountered.

The players:

     macOS v11.6.2
     Bunch v1.4.7
     Bartender 4 v4.1.21
     Karabiner Elements v14.3.0

The set up:

     I have set my “caps lock” as a hyper key via Karabiner.

     I set the hot key for Bunch as “hyper B”.

     I have Bartender configured to “Use Bartender Bar to show hidden items”.

     I am using spaces, and do not have “Displays have separate spaces” set. This is unchecked.

     I have three monitors, and thus the menu bar is only on one monitor.

The issue:

If Bunch is a hidden item in Bartender, when I use the hot key to invoke Bunch it shows not on the monitor with the menu bar, but on the third monitor, and the top several Bunch menu lines are not displayed.

If Bunch is not a hidden item in Bartender, then when I sue the hot key to invoke Bunch it works as expected. This is an acceptable workaround.

Screenshots:

Bunch not hidden in Bartender:

![Bunch 1](https://user-images.githubusercontent.com/95952224/146221084-d63f43fc-9fb2-4d25-b981-ebb21b5b96a6.png)

Bunch hidden in Bartender:

![Bunch 2](https://user-images.githubusercontent.com/95952224/146221186-3ad2ad7a-ce95-4232-988e-1d37de6c72db.png)

Like I noted above, I've set Bunch as unhidden, and this works for me. I will also log an issue with the folks at Bartender.

**Comment by ttscoff** [comment]
Yeah, pretty sure this is a Bartender issue. I don't think Bunch does 
anything that would specifically cause issues. That said, I keep Bunch 
visible in the menu bar when using with Bartender so I haven't 
experienced the issue myself, and most of the menu bar items I hide 
don't have associated keyboard shortcuts that would make me notice 
errant behavior in other apps.

**Reply by SteveMattan** [reply]
Thanks, I figured as much. And likewise with the hidden items and keyboard shortcuts.

Should I hear back from the Bartender folks I'll post an update.

Thanks again and stay well.

**Reply by SteveMattan** [reply]
Good morning.

Here is the reply from Ben at Surtees Studios, maker of Bartender:

-----
Hi Steve, when menu bar items are hidden they are technically pushed off the left side of the menu bar screen, this can happen even without Bartender if the user had more menu bar items than screen width. The issue is when this isn't handled by an app when it displays its UI, unfortunately Bartender cannot correct this for the apps.

But what you could do, is assign the hotkey for Bunch in Bartender instead of in Bunch, by doing that Bartender will make sure Bunch is temporarily visible before activating Bunch.

Thanks.

Ben

-- 
Ben Surtees
Cofounder Surtees Studios
www.surteesstudios.com
-----

As previously noted, I've got Bunch unhidden, and that works for me. 

Stay well, and thanks for Bunch (I almost wrote, "thanks a bunch for Bunch"),

SteveM

**Comment by ttscoff** [comment]
Ooh, I should test that tip out and include it in the docs. Thanks for 
the followup!

-Brett
